packages/proteus/src/main/session/Session.ts

```diff
@@ -329,9 +334,10 @@ export class Session {
     }
 
     encoder.u8(5);
-    encoder.object(Object.keys(this.session_states).length);
+    const sessionStatesIds = Object.keys(this.session_states);
+    encoder.object(sessionStatesIds.length);
 
-    for (const index in this.session_states) {
+    for (const index of sessionStatesIds) {
```

Comment:
```suggestion
    for (const id of sessionStatesIds) {
```

That's not an index anymore but an ID.

Reply:
My naming was wrong, it should've been `sessionStatesIndices`.